sklearn/tree/_tree.pyx

```diff
@@ -993,14 +1229,25 @@ cdef class Splitter:
         safe_realloc(&self.feature_values, n_samples)
         safe_realloc(&self.constant_features, n_features)
 
-        # Initialize y, sample_weight
         self.y = <DOUBLE_t*> y.data
         self.y_stride = <SIZE_t> y.strides[0] / <SIZE_t> y.itemsize
+        
         self.sample_weight = sample_weight
 
     cdef void node_reset(self, SIZE_t start, SIZE_t end,
                          double* weighted_n_node_samples) nogil:
-        """Reset splitter on node samples[start:end]."""
+        """Reset the splitter object to consider samples between start and end.
```

Comment:
I forgot this was partially already documented there. I think what we need is an additional paragraph that explains what are the main methods used to build a decision tree on the three classes of components: Tree, Builder, Splitter, Criterion. Otherwise if it's very hard to get the big picture by reading the code of specific builder / splitter combo.